packages/material-ui-lab/src/TreeItem/TreeItem.js

```diff
@@ -16,38 +16,59 @@ export const styles = theme => ({
     padding: 0,
     outline: 0,
     WebkitTapHighlightColor: 'transparent',
-    '&:focus > $content': {
-      backgroundColor: theme.palette.grey[400],
+    '&:focus > $content $overlay, & > $content $overlay:hover': {
+      opacity: theme.palette.action.hoverOpacity,
     },
+    '&$selected > $content $overlay': {
+      opacity: theme.palette.action.selectedOpacity,
+    },
+    '&$selected > $content $overlay:hover, &$selected:focus > $content $overlay, &:focus > $content $overlay:hover': {
+      opacity: theme.palette.action.hoverOpacity + theme.palette.action.selectedOpacity,
+    },
+  },
+  /* Styles applied to the overlay element. */
+  overlay: {
```

Comment:
I would recommend using the background color instead. I doubt this is an approach we will want to scale to the other components. It both requires an extra DOM node and make overrides confusing. I believe we could try the approach in https://github.com/mui-org/material-ui/blob/c9745319fab39ceee5390de7b9bd8db2f830221a/packages/material-ui-lab/src/PaginationItem/PaginationItem.js#L26.

Reply:
> I doubt this is an approach we will want to scale to the other components.

I had the same experience. This ends up requiring looking at the implementation for customization.